client/lib/viewport/index.js

```diff
@@ -100,51 +100,84 @@ function getMediaQueryList( breakpoint ) {
 	return mediaQueryLists[ breakpoint ];
 }
 
+/**
+ * Returns whether the current window width matches a breakpoint.
+ * @param {String} breakpoint The breakpoint to consider.
+ *
+ * @returns {Boolean} Whether the provided breakpoint is matched.
+ */
 export function isWithinBreakpoint( breakpoint ) {
 	const mediaQueryList = getMediaQueryList( breakpoint );
 	return mediaQueryList ? mediaQueryList.matches : undefined;
 }
 
+/**
+ * Registers a listener to be notified of changes to breakpoint matching status.
+ * @param {String} breakpoint The breakpoint to consider.
+ * @param {Function} listener The listener to be called on change.
+ *
+ * @returns {Function} The function to be called when unsubscribing.
+ */
 export function addWithinBreakpointListener( breakpoint, listener ) {
-	const mediaQueryList = getMediaQueryList( breakpoint );
-
-	if ( mediaQueryList && ! isServer ) {
-		mediaQueryList.addListener( evt => listener( evt.matches ) );
+	if ( ! listener ) {
+		return;
```

Comment:
If we returned `noop` function here, we wouldn't need to check before calling the returned function. More pleasant API to work with.

Reply:
Very good point. Done!